pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -613,11 +629,17 @@ func (c *Controller) getProxyServiceInstancesByPod(pod *v1.Pod, service *v1.Serv
 			log.Warnf("Failed to find port for service %s/%s: %v", service.Namespace, service.Name, err)
 			continue
 		}
-		// There is only one IP for kube registry
+
 		proxyIP := proxy.IPAddresses[0]
 
 		out = append(out, c.getEndpoints(proxyIP, int32(portNum), svcPort, svc))
 
+		// appending the instances by changing the NetworkEndpoint.Address when there are multiple ips
+		for index, ip := range proxy.IPAddresses {
+			if index != 0 {
+				out = append(out, c.overrideServiceIPAndNetwork(out[0], ip))
```

Comment:
This is not right, you donot consider the port.

Reply:
Already modified